### PR TITLE
Port MOM6 to CMake

### PR DESCRIPTION
**Pull request by BrianCurtis-NOAA**
Port MOM6 to Cmake #19

### CODE REVIEW

**Comment by jiandewang** [comment]
Brian:
  I am not familiar with Cmake.
  I had a quick look on your changes, in CMakeLists.txt, SET(lib_ocean_list, seems you manually added all the MOM code here, is that right? can we do something like "src/MOM6/ALE/*" here? The reason I am worried here is that sometimes GFDL may delete several files, or add several files in the new release.

Thanks
Jiande

**Comment by BrianCurtis-NOAA** [comment]
I can try GLOB_RECURSE then for every GFDL change we'd have to re build. I'll do that today.

**Comment by JessicaMeixner-NOAA** [comment]
When we have fully transitioned to cmake, would what you are adding here be all that is left in this repo except for a pointer to the src/MOM6? So no more compile.sh, clean.sh, mkmf, makefile.nuopc?

**Comment by BrianCurtis-NOAA** [comment]
@JessicaMeixner-NOAA I believe that is the goal and how I've structured it. You build everything in "build" directory and when you want to start fresh you just delete the build directory and create it again. CMake isn't supposed to clean up what it's programmed to create only what it uses to build, so keeping everything in the build directory will make it easy to delete and build again. So no need for clean.sh. compile.sh just called mkmf and allowed for three separate outcomes and I've stuck with the first (the one that's enabled and contains all .F90 files in MOM6). The makefile.nuopc seemed to add all files from config_src/nuopc_driver which is the option i've added in the CMake.

@jiandewang GLOB_RECURSE does not follow symlinks (this was disabled a few version ago as it caused issues with long lists) So this means that the files that were symlinked from src/MOM6/pkg (GSW, CVMIX etc...) will not get added to the list. The workaround can be to append to the list manually those files, but then it would have to be updated if anything symlinked to anything new in the future bringing it back to your original problem. Just adding all from src/MOM6/src/pkg seemed to cause more problems because we'd be compiling files we don't need or dependencies i would have to search for.

I will move towards the hybrid option of using GLOB_RECURSE for most files and appending a static list of those files that were symlinked. It will, at the very least, make the CMakeLists.txt more readable.

Thanks for your comments. I will get those changes in ASAP.

File: cmake/Intel.cmake [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -xHOST -qno-opt-dynamic-align")
+    endif()
+endif()
+
+if(REPRO)
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O2 -debug minimal -fp-model consistent -qoverride-limits -g -traceback")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2 -debug minimal")
+else()
+    if(DEBUG)
+        set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -g -O0 -check -check noarg_temp_created -check nopointer -warn -warn noerrors -fp-stack-check -fstack-protector-all -fpe0 -debug -traceback -ftrapuv")
```
Comment (DeniseWorthen) [inline cmake/Intel.cmake]: We lost our "-link_mpi=debug" flag again.

Comment (BrianCurtis-NOAA) [reply cmake/Intel.cmake]: Thanks!

https://github.com/ufs-community/ufs-weather-model/blob/develop/cmake/Intel.cmake is what I had used. I will update this version, but do we address it with the ufs-community/ufs-weather-model as well or i'm assuming somewhere there's a branch that addresses this?

Comment (DeniseWorthen) [reply cmake/Intel.cmake]: I think that for the uncoupled model it is perhaps not crucial for debugging. But for the coupled model it appears to make a big difference in creating error messages that correctly point to the component model where the debug options are triggering a fault. Our experience was that w/o this flag, the model would fault in some random location (usually the ice model).

**Comment by BrianCurtis-NOAA** [comment]
New changes submitted from yesterdays discussions. commit c950eb6

**Comment by BrianCurtis-NOAA** [comment]
I believe this was completed a long time ago. Closing.

**Pull request closed without merging**